packages/aws-cdk-lib/core/lib/stack.ts

```diff
@@ -1047,7 +1075,20 @@ export class Stack extends Construct implements ITaggable {
         Annotations.of(this).addInfo(`Number of resources: ${numberOfResources} is approaching allowed maximum of ${this.maxResources}`);
       }
     }
-    fs.writeFileSync(outPath, JSON.stringify(template, undefined, 1));
+
+    const indent = this._suppressTemplateIndentation ? undefined : 1;
+    const templateData = JSON.stringify(template, undefined, indent);
+
+    if (templateData.length > TEMPLATE_BODY_MAXIMUM_SIZE) {
+      throw new Error(`Template size for stack '${this.node.path}' exceeds limit: ${templateData.length}/${TEMPLATE_BODY_MAXIMUM_SIZE}`);
```

Comment:
Lets change this to a warning too. It will still fail fast on deployment.

Reply:
ok